File: solutions/javascript/binary-search/1/binary-search.js
/**
 * Performs a binary search on a sorted array to find the index of a target value.
 * @param {number[]} sortedArray - A sorted array of numbers in ascending order.
 * @param {number} targetValue - The number to search for in the array.
 * @returns {number} The index of the targetValue in the array.
 * @throws {Error} If the targetValue is not found in the array.
 */
export const find = (sortedArray, targetValue) => {
  let left = 0;
  let right = sortedArray.length - 1;
  let mid;

  while (left <= right) {
    mid = Math.floor((left + right) / 2);
    if (sortedArray[mid] === targetValue) return mid;
    else if (sortedArray[mid] < targetValue) left = mid + 1;
    else right = mid - 1;
  }
  throw new Error('Value not in array');
};

File: solutions/lua/binary-search/1/binary-search.lua
return function(array, target)
  local high, low = #array, 1
  while low <= high do
    local mid = math.floor((low + high) / 2)
    if array[mid] == target then
      return mid
    elseif array[mid] < target then
      low = mid + 1
    else
      high = mid - 1
    end
  end
  return -1
end